### PR TITLE
chore(cli): Enable 'next' and 'experimental' tags for `rw upgrade`

### PR DESCRIPTION
**Pull request by Josh-Walker-GM**
**Problem**
We want to allow users to select the `next` and `experimental` tags when running `yarn rw upgrade --tag [tag]` but this is currently not allowed.

**Changes**
1. Add `next` and `experimental` to the allowed tags list.
2. Update the docs to reflect the updated tag options.

### CODE REVIEW

File: docs/docs/cli-commands.md
Comment (Josh-Walker-GM) [inline docs/docs/cli-commands.md:2046]: Need to also warn of "next" in the "WARNING: Unstable releases..." section?

**Comment by Josh-Walker-GM** [comment]
Reverted to simply add "next" and "experimental" to the allowed list of tags. 

I don't understand why we have an "experimental" tag given that we expect users to simply use the full SEMVER version for different experimental branches/features.

File: docs/docs/cli-commands.md
Comment (thedavidprice) [reply docs/docs/cli-commands.md:2046]: yes!

File: packages/cli/src/commands/upgrade.js
Comment (thedavidprice) [inline packages/cli/src/commands/upgrade.js:33]:
```suggestion
        '[choices: "rc", "canary", "latest", "next", "experimental", or a specific-version (see example below)] WARNING: "canary", "rc" and "experimental" are unstable releases! And "canary" releases include breaking changes often requiring codemods if upgrading a project.',
```

**Comment by Tobbe** [comment]
> I don't understand why we have an "experimental" tag given that we expect users to simply use the full SEMVER version for different experimental branches/features.

Let's discuss this on the next core team meeting before we merge this PR please 🙂

File: docs/docs/cli-commands.md
Comment (thedavidprice) [inline docs/docs/cli-commands.md:2046]:
```suggestion
| `--tag, -t`     | Choices are "rc", "canary", "latest", "next", "experimental", or a specific version (e.g. "0.19.3"). WARNING: Unstable releases in the case of "canary", "rc", "next", and "experimental". And "canary" releases include breaking changes often requiring codemods if upgrading a project. |
```

**Review by thedavidprice — APPROVED** [review]
This looks great @Josh-Walker-GM I made some copy changes. Could you do a final check/confirmation and then merge?

Looping in @jtoar if we want to cherry-pick this into v4.5-RC

**Comment by thedavidprice** [comment]
@Tobbe I'm asking to merge this before next meeting. Happy to discuss further.

**Comment by Josh-Walker-GM** [comment]
I checked again locally and things looked and worked fine. I'll enable auto-merge but this might get stuck on the flaky logger CI error for a little bit until that's resolved separately.

**Pull request merged**
